Fixed 2 bok choy tests failing on chrome

raeeschachar opened the pull request:
test_undo_changes and test_manaul_change in test_studio_settings.py
@benpatterson @jzoldak 
Please Review.
I still can't figure out the reason why Chrome won't reset Course Announcement Date field back to its default value but changing the field to Course Advertised Start Date (which inputs/outputs the same values like Course Announcement Date field) completes our test successfully. There might be an issue with the undo button. Se Won Jang was the author but can't find him to tag for review.

benpatterson commented:
jenkins run bokchoy

benpatterson commented:
@raeeschachar I'm just curious if you've tried this on chrome in stage? (E.g., is it an actual product bug?)

raeeschachar commented:
jenkins run bokchoy

raeeschachar commented:
@benpatterson that was the first thing I checked, works fine and displays 'null' after clicking 'undo' button.

benpatterson commented:
Gotcha. :+1: 

CC @andy-armstrong @cahrens FYI on this slight test change. If the distinction between these fields is particularly important, we may need your help getting them working on chrome.

cahrens commented:
The bok choy test is still failing.

benpatterson commented:
jenkins run bokchoy

cahrens commented:
Isn't the test that failed one related to the code change (I'm nervous about just re-running the tests).

raeeschachar commented:
@cahrens @benpatterson got it, will send the correction in a bit

raeeschachar commented:
@cahrens @benpatterson 
Just found out this is a bug. Course Advertised Start Date shouldn't accept string format. http://screencast.com/t/mhmOVvxcc

@aqdasrehman We should explicitly test all the fields on the Advanced Settings page (i.e their inputs, outputs and complete field validation) and add a Jira ticket for all those that are failing.

raeeschachar commented:
@benpatterson @jzoldak @cahrens 
Need help with this one. If the fields are reduced to 4 with any combination of tests, values are reverted back to default and works on chrome but if they are 5 (tested with maximum waits), values returned are not changed to default (waited for forever). One option could be, we can test it by 2 groups of combinations with 4 fields each. Can there be something else to fix this?

raeeschachar commented:
Do we need to upgrade Chrome? Current version is 45 and the chrome bok choy tests run on 30.

benpatterson commented:
@raeeschachar upgrading to chrome would be great, but that is a more drawn out process. I'm hoping Adnan can help with that when he arrives.

raeeschachar commented:
@benpatterson @jzoldak @cahrens 
Please review again :)
Undo Changes/ Change Manually buttons were not clicking (took some time to find that out) as they were hiding behind "Studio's having trouble saving your work" notification. Added scroll to both so that on browsers like Chrome, buttons can be clicked. 
And changed the Course Announcement Date field back (which earlier I thought was the issue). Thanks

raeeschachar commented:
jenkins run js

jzoldak commented:
LGTM :+1:

cahrens commented:
:+1:

benpatterson commented on common/test/acceptance/pages/studio/settings_advanced.py:
```diff
@@ -52,7 +52,9 @@ def undo_changes_via_modal(self):
         Wait for the undoing process to load via ajax call.
+        Before that Scroll so the button is clickable on all browsers
         """
+        self.browser.execute_script("window.scrollTo(0, 500)")
```
I'm wondering if you can get the coordinates first, then feed those coordinates into the script?

That way if the layout changes, the tests will be resilient. Plus, that pattern can be adopted elsewhere; I had seen this issue on other tests a few months ago...

jzoldak replied:
You also might be able to use scrollIntoView.
See [this stack exchange post](http://stackoverflow.com/questions/3401343/scroll-element-into-view-with-selenium).

raeeschachar commented:
@benpatterson @jzoldak 
Will this work? Please review.

raeeschachar commented:
jenkins run lettuce

benpatterson commented:
Solid. :+1:

jzoldak commented:
:+1:
